### PR TITLE
[bazel] add slot management to `opentitan_functest` macro

### PR DESCRIPTION
**Pull request by timothytrippel**
This adds support to the `opentitan_functest` macro to specify which
stage and slot an `opentitan_flash_binary` gets built for, defaulting to
the `silicon_creator` stage, slot A.

This partially addresses #10712.

This functionality will soon be expanded to address building more complex
test cases with multiple stages / slots filled, to address #13511.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

File: rules/opentitan_test.bzl [hunk 1]
```diff
@@ -222,6 +230,7 @@ def opentitan_functest(
         test_in_rom = False,
         ot_flash_binary = None,
         signed = False,
+        slot = "silicon_creator_a",
```
Comment (alphan) [inline rules/opentitan_test.bzl:233]: nit: can we make "virtual" (the setting that we will use most until ePIC is ready) the default?

File: sw/device/lib/testing/test_framework/BUILD [hunk 2]
```diff
@@ -143,36 +144,6 @@ cc_library(
     ],
 )
 
-cc_library(
```
Comment (alphan) [inline sw/device/lib/testing/test_framework/BUILD:146]: nice!

**Review by alphan — APPROVED** [review]

File: rules/opentitan_test.bzl (hunk 1 above)
Comment (timothytrippel) [reply rules/opentitan_test.bzl]: sure thing! done.

Comment (timothytrippel) [reply rules/opentitan_test.bzl:250]: actually (as we discussed offline) need to keep the default slot to the virtual slot, b/c address translation is not enabled in the test ROM yet. once it is added, we can make the default slot the virtual slot. Tracking in #10712

**Pull request merged**